sdk/data_tables/src/clients/entity_client.rs

```diff
@@ -102,10 +102,11 @@ impl EntityClient {
         &self,
         url: Url,
         method: Method,
+        headers: Headers,
         request_body: Option<Bytes>,
     ) -> azure_core::Result<Request> {
         self.partition_key_client
-            .prepare_request(url, method, request_body)
```

Comment:
Since all of these `prepare_request` methods need to be called with the final set of headers, maybe renaming them to `finalize_request` would be better. That way it's more clear that this should be called *last*.